src/server/replica.cc

```diff
@@ -431,43 +432,40 @@ error_code Replica::InitiateDflySync() {
     shard_flows_[i].reset(new Replica(master_context_, i, &service_));
   }
 
-  SyncBlock sb{num_df_flows_};
+  // Allocate shared, because the error handler might outlive the function scope.
+  auto sync_block = std::make_shared<SyncBlock>();
 
-  AggregateError ec;
+  auto err_handler = [this, sync_block](const auto& ge) {
+    sync_block->Add(num_df_flows_);  // Unblock sync_block.
+    DefaultErrorHandler(ge);         // Close sockets to unblock flows.
+  };
+  if (cntx_.Switch(std::move(err_handler)))
+    return cntx_;
+
+  // Start full sync flows.
   auto partition = Partition(num_df_flows_);
   shard_set->pool()->AwaitFiberOnAll([&](unsigned index, auto*) {
     for (auto id : partition[index]) {
-      if ((ec = shard_flows_[id]->StartFullSyncFlow(&sb)))
-        break;
+      auto ec = shard_flows_[id]->StartFullSyncFlow(sync_block.get(), &cntx_);
+      if (ec)
+        cntx_.Error(ec);
```

Comment:
Doing this in one line would also be great 😅
Either we let `Error` ignore empty errors, or we use some kind of operator, but I can't find one, because `=` is really not obvious.

Reply:
rename it to `PassErrorStatus` and then it's more clear that you may pass an empty error